### PR TITLE
[Quest] Convert Making Amends to IF

### PR DESCRIPTION
**Pull request by jamesbradleym**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Converts `Making Amends` to IF

## Steps to test these changes

Pre-Reqs:
Windy fame 2+
Horutoto Ruins Experiment Mission not active

Start NPC:
Hakkuru-Rinkuru : !pos -111 -4 101 240

Trade Hakkuru-Rinkuru Animal Glue (!additem 937)

### CODE REVIEW

File: scripts/quests/windurst/Making_Amends.lua [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    fame     = 75,
+    fameArea = xi.fameArea.WINDURST,
+    title    = xi.title.QUICK_FIXER,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            local horutotoRuinsExperimentStatus = player:getMissionStatus(xi.mission.log_id.WINDURST, xi.mission.id.windurst.THE_HORUTOTO_RUINS_EXPERIMENT)
```
Comment (claywar) [inline scripts/quests/windurst/Making_Amends.lua]: Just noting this now, but missionStatus will be returned regardless of nation.  You're going to need to be more specific

Comment (jamesbradleym) [reply scripts/quests/windurst/Making_Amends.lua]: I might be misunderstanding the mechanics at play here, so pardon my ignorance.

As you note - this checks regardless of nation, but should still return `0` for `player:getMissionStatus` regardless of nation if you aren't actually on `THE_HORUTOTO_RUINS_EXPERIMENT`, right? In that case this check does the right thing. I suppose I could add a nation check and make this an or for not Windurst or not on `THE_HORUTOTO_RUINS_EXPERIMENT`?

Comment (cocosolos) [reply scripts/quests/windurst/Making_Amends.lua]: Generally, `getMissionStatus` only takes one argument, which is the log you're checking (there are exceptions e.g. CoP). `player:getMissionStatus(xi.mission.log_id.WINDURST)` would return whatever the mission status variable is for whatever your current mission is for Windurst, and 0 if you're any other nation or no current mission.

Comment (cocosolos) [reply scripts/quests/windurst/Making_Amends.lua]: For this check, you don't want to be checking the mission status, but `player:getCurrentMission(xi.mission.log_id.WINDURST) ~= xi.mission.id.windurst.THE_HORUTOTO_RUINS_EXPERIMENT` instead.

Comment (cocosolos) [reply scripts/quests/windurst/Making_Amends.lua]: You also want to check `if player:getCurrentMission(xi.mission.log_id.WINDURST) == xi.mission.id.windurst.SAINTLY_INVITATION and player:getMissionStatus(xi.mission.log_id.WINDURST) > 1` to verify they're not blocked by the Saintly Invitation BCNM.

Comment (jamesbradleym) [reply scripts/quests/windurst/Making_Amends.lua]: Thanks @cocosolos 🙏🏼

**Comment by jamesbradleym** [comment]
Closing this PR to reduce open PRs and prevent staleness—will revisit.

**Pull request closed without merging**